### PR TITLE
Refactoring for `Adding support for Redirect Token Calls fro evm-module` PR

### PR DESCRIPTION
**Pull request by nikolovyanko**
This is a follow-up refactoring PR for [PR](https://github.com/hashgraph/hedera-services/pull/4880).

Changes in the PR:

- The logic from services' `UpdateTrackingLedgerAccount ` and hedera-evm's `UpdatedHederaEvmAccount ` moved to a common [UpdateTrackingAccount](https://github.com/hashgraph/hedera-services/blob/8723c8c1db616e98d468e85a89fb5b13fa906895/hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccount.java) in the hedera-evm package.
- Common logic for account tracking moved from `AbstractLedgerWorldUpdater ` to the parent `AbstractLedgerEvmWorldUpdater ` for shared usage.
- An abstraction for account properies tracking introduced via `UpdatedAccountTracker ` and services implementation of it `UpdatedAccountTrackerImpl`.

The workflow in this PR is being test from mirror-node side in [PR](https://github.com/hashgraph/hedera-mirror-node/pull/5424l)

Refactors https://github.com/hashgraph/hedera-services/issues/4798

### CODE REVIEW

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccount.java [hunk 1]
```diff
@@ -37,35 +33,50 @@
...
-    private TransactionalLedger<AccountID, AccountProperty, HederaAccount> trackingAccounts;
-
+public class UpdateTrackingAccount<A extends Account> implements MutableAccount, EvmAccount {
+    private final Address address;
+    private final Hash addressHash;
+    private long nonce;
+    private Wei balance;
+    private HederaEvmEntityAccess hederaEvmEntityAccess;
     private boolean storageWasCleared = false;
+    private final UpdatedAccountTracker trackingAccounts;
```
Comment (IvanKavaldzhiev) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccount.java:46]: Probably we can rename this field to `updatedAccountTracker`

File: hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccountTest.java [hunk 2]
```diff
@@ -197,28 +175,20 @@ void canClearStorage() {
...
-    @Test
-    void setBalancePropagatesToUsableTrackingAccounts() {
-        final var account = new WorldStateAccount(targetAddress, Wei.of(initialBalance), codeCache, entityAccess);
-        final var subject = new UpdateTrackingLedgerAccount<>(account, trackingAccounts);
-
-        subject.setBalance(Wei.of(Long.MAX_VALUE));
-        assertEquals(Long.MAX_VALUE, subject.getBalance().toLong());
-        verify(trackingAccounts).set(targetId, AccountProperty.BALANCE, Long.MAX_VALUE);
-    }
+    //    void setBalancePropagatesToUsableTrackingAccounts() {
```
Comment (IvanKavaldzhiev) [inline hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccountTest.java:185]: This is probably no more necessary.

File: hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/store/models/UpdatedHederaEvmAccountTest.java [hunk 3]
```diff
@@ -41,29 +41,28 @@ class UpdatedHederaEvmAccountTest {
...
-    private UpdatedHederaEvmAccount subject;
+    private UpdateTrackingAccount subject;
 
     @Mock
     private HederaEvmEntityAccess hederaEvmEntityAccess;
 
     @BeforeEach
     void setUp() {
-        subject = new UpdatedHederaEvmAccount(address);
+        subject = new UpdateTrackingAccount(address, null);
```
Comment (IvanKavaldzhiev) [inline hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/store/models/UpdatedHederaEvmAccountTest.java:51]: On all places where we pass null, we can pass a mock for `UpdateAccountTracker`

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.evm.store;
+
+import org.hyperledger.besu.datatypes.Address;
+
+public interface UpdatedAccountTracker {
```
Comment (IvanKavaldzhiev) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java:21]: Probably we can rename it to `UpdateAccountTracker`

Comment (tinker-michaelj) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java:21]: Perhaps we could add some javadoc. A first draft might be,
```
/**
* When an EVM call changes an account's balance, this change needs to be immediately visible
* to any Hedera system contract that runs within the same frame. (I.e., we cannot wait for commit()
* to be called on a WorldUpdater; we need to immediately update the "parallel" Hedera world state).
* 
* <p>This type defines the callback a mutated account uses to report its balance changes to 
* the Hedera world state when running an EVM transaction in a consensus node. (Accounts
* "mutated" during an eth_estimateGas call running on a mirror node don't actually change, so
* we can provide a null implementation of the callback there.)
* */
```

**Review by tinker-michaelj — APPROVED** [review]
This all makes sense to me! 🥂 I suggested a first draft of some javadoc for the `UpdatedAccountTracker` interface, and there are probably a few other places we might sprinkle something in.

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.evm.store;
+
+import org.hyperledger.besu.datatypes.Address;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java:20]: Please add javadocs for interface and its methods

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/AbstractLedgerEvmWorldUpdater.java [hunk 6]
```diff
@@ -131,13 +135,22 @@ public EvmAccount getAccount(Address address) {
...
 
-    private UpdatedHederaEvmAccount<A> track(final UpdatedHederaEvmAccount<A> account) {
-        account.setEvmEntityAccess(hederaEvmEntityAccess);
-        updatedEvmAccounts.put(account.getAddress(), account);
+    public Map<Address, UpdateTrackingAccount<A>> getUpdatedAccounts() {
+        return updatedAccounts;
+    }
+
+    // FeatureWork public ContractAliases aliases()
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/AbstractLedgerEvmWorldUpdater.java:149]: Please add javadocs

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccount.java [hunk 7]
```diff
@@ -37,35 +34,52 @@
 import org.hyperledger.besu.evm.ModificationNotAllowedException;
 import org.hyperledger.besu.evm.account.Account;
 import org.hyperledger.besu.evm.account.AccountStorageEntry;
+import org.hyperledger.besu.evm.account.EvmAccount;
 import org.hyperledger.besu.evm.account.MutableAccount;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdateTrackingAccount.java:39]: same here

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/UpdatedAccountTrackerImpl.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import static com.hedera.node.app.service.mono.utils.EntityIdUtils.accountIdFromEvmAddress;
+
+import com.hedera.node.app.service.evm.store.UpdatedAccountTracker;
+import com.hedera.node.app.service.mono.ledger.TransactionalLedger;
+import com.hedera.node.app.service.mono.ledger.properties.AccountProperty;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hederahashgraph.api.proto.java.AccountID;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hyperledger.besu.datatypes.Address;
+
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/UpdatedAccountTrackerImpl.java:29]: same here

**Review by Nana-EC — APPROVED** [review]
LG.
Just add javadocs as others have noted

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/AbstractLedgerEvmWorldUpdater.java (hunk 6 above)
Comment (nikolovyanko) [reply hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/AbstractLedgerEvmWorldUpdater.java:165]: Added.

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java (hunk 4 above)
Comment (nikolovyanko) [reply hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/UpdatedAccountTracker.java]: Added.

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by stoqnkpL — APPROVED** [review]

**Pull request merged**